x/smart/stream_dialer.go

```diff
@@ -250,21 +250,31 @@ func (f *StrategyFinder) testDialer(ctx context.Context, dialer transport.Stream
 		testAddr := net.JoinHostPort(testDomain, "443")
 		f.logCtx(ctx, "🏃 running test: '%v' (domain: %v)\n", transportCfg, testDomain)
 
-		ctx, cancel := context.WithTimeout(ctx, f.TestTimeout)
+		testCtx, cancel := context.WithTimeout(ctx, f.TestTimeout)
 		defer cancel()
-		testConn, err := dialer.DialStream(ctx, testAddr)
+
+		timeoutChan := make(chan struct{})
+		go func() {
+			<-testCtx.Done()
+			if errors.Is(testCtx.Err(), context.DeadlineExceeded) {
+				f.logCtx(testCtx, "⏱️ dialer failure, test timed out: '%v' (domain: %v), duration=%v ❌\n", transportCfg, testDomain, time.Since(startTime))
```

Comment:
I'd rather stick to 🏁 to indicate test completion for consistency.

Reply:
done